src/main/java/io/cdap/plugin/switchcase/route/BasicRoutingFunctions.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.switchcase.route;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Class that contains various implementations of {@link BasicRoutingFunction}
+ */
+final class BasicRoutingFunctions {
+
+  /**
+   * Routing function based on equality
+   */
+  static class EqualsFunction implements BasicRoutingFunction {
+
+    @Override
+    public boolean evaluate(String actualValue, String compareValue) {
+      return compareValue.equals(actualValue);
+    }
+  }
+
+  /**
+   * Routing function based on non-equality
+   */
+  static class NotEqualsFunction implements BasicRoutingFunction {
+
+    @Override
+    public boolean evaluate(String actualValue, String compareValue) {
+      return !compareValue.equals(actualValue);
+    }
+  }
+
+  /**
+   * Routing function based on substrings
+   */
+  static class ContainsFunction implements BasicRoutingFunction {
+
+    @Override
+    public boolean evaluate(String actualValue, String compareValue) {
+      return actualValue.contains(compareValue);
+    }
+  }
+
+  /**
+   * Routing function based on non-substrings
+   */
+  static class NotContainsFunction implements BasicRoutingFunction {
+
+    @Override
+    public boolean evaluate(String actualValue, String compareValue) {
+      return !actualValue.contains(compareValue);
+    }
+  }
+
+  /**
+   * Routing function based on presence in a list of possible values
+   */
+  static class InFunction implements BasicRoutingFunction {
+
+    @Override
+    public boolean evaluate(String actualValue, String compareValue) {
+      List<String> possibleValues = Arrays.asList(compareValue.split("\\|"));
```

Comment:
Why doesn't the caller pass parsed value ?

Reply:
the contract is defined at the interface as string for both parameters, and this is the way that it will be passed from the UI as well. The individual ``Function`` implementations can then do the parsing based on their needs.